addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -474,6 +477,104 @@ def test_mailing_shortener(self):
                     link_params=link_params,
                 )
 
+    @users('user_marketing')
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mailing_launch_flow(self):
+        """ Test that mail should be sent directly if total recipients are less than value
+        specified by config param 'mass_mailing_immediate_send_nbr', otherwise it should
+        enqueue the mailing """
+        res_partners = self._create_partner()
+        # number of emails are less then specified limit
+        mailing_1 = self.env['mailing.mailing'].create({
+            'name': 'TestMailingOne',
+            'subject': 'One',
+            'body_html': 'This is mass mail marketing demo',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'mailing_domain': [('id', 'in', res_partners[0:3].ids)]
+        })
+
+        # number of emails are more then specified limit
+        mailing_2 = self.env['mailing.mailing'].create({
+            'name': 'TestMailingTwo',
+            'subject': 'Two',
+            'body_html': 'This is mass mail marketing demo',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'mailing_domain': [('id', 'in', res_partners.ids)]
+        })
+        with self.mock_mail_gateway(mail_unlink_sent=False), patch.object(MassMailing, '_get_mass_mailing_immediate_send_nbr', return_value=3):
+            (mailing_1 | mailing_2).action_launch()
+        self.assertEqual(mailing_1.state, 'done')
+        self.assertEqual(mailing_2.state, 'in_queue')
+        self.assertEqual(len(self._new_mails.exists()), 3)
+        self.assertEqual(len(self._mails), 3)
+        # test that mailing_1 mails are sent only
+        for partner in res_partners[0:3]:
```

Comment:
Hello @std-odoo ,
In assertmailmail when we give multiple recipients, then it should be present in all mails. As it is checking that the recipients are there in all mails but we will have single recipient for single mail. So, it will fail due to which we need loop here.